install/ui/src/freeipa/widgets/MigrateScreen.js

```diff
@@ -0,0 +1,215 @@
+//
+// Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+//
+
+define(['dojo/_base/declare',
+    'dojo/dom-construct',
+    'dojo/topic',
+    '../ipa',
+    '../config',
+    '../text',
+    '../util',
+    './LoginScreenBase'
+    ],
+    function(declare, construct, topic, IPA, config, text, util,
+        LoginScreenBase) {
+
+        /**
+         * Widget with password migration form.
+         *
+         * Supported operations:
+         *
+         * - password migration
+         *
+         * @class widgets.MigrateScreen
+         */
+        var MigrateScreen = declare([LoginScreenBase], {
+
+            migration_error_msg: "There was a problem with your request."+
+            "Please, try again later.",
+
+            migration_failure_msg: "Password migration was not successful",
+
+            migration_info_msg: "<h1>Password Migration</h1><p>"+
+            "If you have been sent here by your administrator, your personal "+
+            "information is being migrated to a new identity management "+
+            "solution (IPA).</p><p>Please, enter your credentials in the form"+
+            " to complete the process. Upon successful login your kerberos "+
+            "account will be activated.</p>",
+
+            migration_invalid_password: "The password or username you entered"+
+            " is incorrect",
+
+            migration_success: "Password migration was successful",
+
+            //nodes:
+            migrate_btn_node: null,
+
+            render_buttons: function(container) {
+                this.migrate_btn_node = IPA.button({
+                    label: text.get('@i18n:buttons.migrate', "Migrate"),
+                    'class': 'btn-primary btn-lg',
+                    click: this.on_confirm.bind(this)
+                })[0];
+                construct.place(this.migrate_btn_node, container);
+            },
+
+            post_create_fields: function() {
```

Comment:
Both username and password should be required fields all the time, so there is no need to set required property of the fields dynamically. I.e., this method can be removed.